pkg/model/firewall.go

```diff
@@ -521,3 +403,44 @@ func JoinSuffixes(src SecurityGroupInfo, dest SecurityGroupInfo) string {
 
 	return s + d
 }
+
+func (b *KopsModelContext) AddDirectionalGroupRule(c *fi.ModelBuilderContext, t *awstasks.SecurityGroupRule) {
+
+	name := generateName(t)
+	t.Name = fi.String(name)
+
+	klog.V(8).Infof("Adding rule %v", name)
+	c.AddTask(t)
+
+}
+
+func generateName(o *awstasks.SecurityGroupRule) string {
+
+	var target, dst, src, direction, proto string
+	if o.SourceGroup != nil {
+		target = fi.StringValue(o.SourceGroup.Name)
+	} else if o.CIDR != nil && fi.StringValue(o.CIDR) != "" {
+		target = fi.StringValue(o.CIDR)
+	} else {
+		target = "0.0.0.0/0"
+	}
+
+	if o.Protocol == nil || fi.StringValue(o.Protocol) == "" {
+		proto = "AllProtos"
```

Comment:
Should be enough to say "all".
```suggestion
		proto = "all"
```

Reply:
all what? :) I used allprotos because for descriptiveness when one reads the names in tasks, logs etc.
But I don't have a strong opinion.

Reply:
I guess it could be replaced with a `-` or something else, but the location will be used for protocol everywhere.